src/combinatorics/simple_graph/prod.lean

```diff
@@ -50,6 +50,17 @@ by rw [box_prod_adj, and_iff_left rfl, or_iff_left (λ h : H.adj b b ∧ _, h.1.
 @[simp] lemma box_prod_adj_right : (G □ H).adj (a, b₁) (a, b₂) ↔ H.adj b₁ b₂ :=
 by rw [box_prod_adj, and_iff_left rfl, or_iff_right (λ h : G.adj a a ∧ _, h.1.ne rfl)]
 
+lemma box_prod_neighbor_set (x : α × β) :
+  (G □ H).neighbor_set x = ((λ a, (⟨a,x.2⟩ : α × β)) '' (G.neighbor_set x.1)) ∪
+                           ((λ b, (⟨x.1,b⟩ : α × β)) '' (H.neighbor_set x.2)) :=
+begin
+  ext ⟨a',b'⟩,
+  simp only [mem_neighbor_set, box_prod_adj, set.mem_union_eq, set.mem_image, prod.mk.inj_iff,
+             exists_eq_right_right],
+  simp_rw [and_comm _ (x.2 = b'), exists_eq_right_right],
+  finish,
+end
```

Comment:
I suggest formulating this lemma as
```lean
lemma box_prod_neighbor_set (x : α × β) :
  (G □ H).neighbor_set x = (G.neighbor_set x.1) ×ˢ {x.2} ∪ {x.1} ×ˢ (H.neighbor_set x.2) :=
begin
  ext ⟨a', b'⟩,
  simp only [mem_neighbor_set, set.mem_union, box_prod_adj, set.mem_prod, set.mem_singleton_iff],
  simp only [eq_comm, and_comm],
end
```

Reply:
(I forgot to mention that one reason for the change in proof is that `finish` is not accepted in mathlib anymore since there are no plans at the moment to port it to Lean 4.)